grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -150,6 +152,25 @@ public String toString() {
   private RoundRobinPicker currentPicker =
       new RoundRobinPicker(Collections.<DropEntry>emptyList(), Arrays.asList(BUFFER_ENTRY));
 
+  private final SubchannelStateListener subchannelStateListener = new SubchannelStateListener() {
```

Comment:
One thing you should consider:  Make these listeners a named private class.  It makes debugging much easier in a stack trace.  Also, the named classes can be made final, which avoids letting Mockito.spy extend them.